projects/fsastorefrontlib/src/cms-components/user-profile/user-profile.component.ts

```diff
@@ -58,13 +81,70 @@ export class UserProfileComponent implements OnInit, OnDestroy {
   assets: { [key: string]: any }[];
   assetSelected: AssetTableType;
   showAddressForm = false;
+  showProductOverview = true;
   userId: string;
   customerId: string;
 
+  selectedProducts = new BehaviorSubject(ProductOverviewCategory.ALL);
+  selectedProducts$ = this.selectedProducts.asObservable();
+
+  allProducts$: Observable<CmsComponent[]>;
+  productsCountByCategory$: Observable<{
+    [key: string]: typeof ProductOverviewCategory;
+  }>;
+  filteredProducts$: Observable<CmsComponent[]>;
+
   ngOnInit(): void {
+    this.allProducts$ = this.componentData.data$.pipe(
+      map(data => data.children.split(' ')),
+      switchMap(children =>
+        from(children).pipe(
+          mergeMap(child => this.cmsService.getComponentData(child)),
+          take(children.length),
+          toArray()
+        )
+      )
+    );
```

Comment:
Maybe move this into separate method that returns this observable. `ngOnInit` should contain only calls to methods when possible.